beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -6559,6 +6554,70 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
         }
     }
 
+    pub fn with_proposer_cache<V, E: From<BeaconChainError> + From<BeaconStateError>>(
```

Comment:
This forces the caller to implement `From` for both error types, it feels a bit strange but given this works now, i think it's fine to leave it as it is.